Lib/dataclasses.py

```diff
@@ -990,7 +997,7 @@ def _process_class(cls, init, repr, eq, order, unsafe_hash, frozen):
     if not getattr(cls, '__doc__'):
         # Create a class doc-string.
         cls.__doc__ = (cls.__name__ +
-                       str(inspect.signature(cls)).replace(' -> None', ''))
+                       str(inspect.signature(cls)).replace(' -> NoneType', ''))
```

Comment:
Ouch, it's really too bad that inspect returned `NoneType` here. :-(

Reply:
Hm, that comes from `get_type_hints()`. And it can occur in other positions as well, e.g.
```py
>>> @dataclass
... class C:
...   a: None
...   b: typing.List[None]
... 
>>> import typing
>>> C.__doc__
'C(a: None, b: List[NoneType])'
>>> 
```

Reply:
Yeah, noticed that and was trying to figure out why there is such a condition;
https://github.com/python/cpython/blob/619f98045d73859688e5cfdb946d183e6d493a64/Lib/typing.py#L1408-L1409

Any hints regarding that?

Reply:
Well, hm... inspect.Signature objects have always had "NoneType" for certain situations. Let's say that's a separate bug. In a sense the new situation is more consistent because it *always* has NoneType. According to PEP 484 it should always have None. But I don't know if we can change that without breakage.

Reply:
Sorry, my previous comment was in response to myself -- I hadn't seen your reply yet. Now responding to this:

> Yeah, noticed that and was trying to figure out why there is such a condition;
> 
> https://github.com/python/cpython/blob/619f98045d73859688e5cfdb946d183e6d493a64/Lib/typing.py#L1408-L1409
> 
> Any hints regarding that?

Apparently get_type_hints() tries to be careful to always return annotations that are types, hence the change from None to NoneType. This is old behavior so should be preserved (though it is undocumented -- it's not mentioned in Doc/library/typing.rst nor in PEP 484).

As I said I think for inspect.signature it is debatable, but I think we should file a separate bug and for now keep the workaround as you now have it.